src/Evented.ts

```diff
@@ -117,36 +44,48 @@ export function isGlobMatch(globString: string | symbol, targetString: string |
 	}
 }
 
+export type EventedCallback<T = EventType, E extends EventObject<T> = EventObject<T>> = {
+	/**
+	 * A callback that takes an `event` argument
+	 *
+	 * @param event The event object
+	 */
+
+	(event: E): boolean | void;
+};
+
+/**
+ * A type which is either a targeted event listener or an array of listeners
+ * @template T The type of target for the events
+ * @template E The event type for the events
+ */
+export type EventedCallbackOrArray<T = EventType, E extends EventObject<T> = EventObject<T>> = EventedCallback<T, E> | EventedCallback<T, E>[];
+
 /**
  * Event Class
  */
-export class Evented extends Destroyable implements Evented {
+export class Evented<M extends {} = {}, T = EventType, O extends EventObject<T> = EventObject<T>> extends Destroyable {
 
-	/**
-	 * map of listeners keyed by event type
-	 */
-	protected listenersMap: Map<string, EventedCallback<EventObject>> = new Map<string, EventedCallback<EventObject>>();
+	// The following member is purely so TypeScript remembers the type of `M` when extending so
+	// that the utilities in `on.ts` will work
+	// tslint:disable-next-line
+	protected __typeMap__?: M;
```

Comment:
This seems is a bit hacky :(

Reply:
Yeah, feels _dirty_ to me too... 😢

Reply:
Yeah, it's dirty, but I'm not sure if there's any other way. TypeScript isn't able to remember the type map for use with the stand-alone `on()` function without it.

Reply:
I've filed [an issue](https://github.com/Microsoft/TypeScript/issues/20348) with a reduced test case outlining this problem. It feels like a bug.